### PR TITLE
Javascript carousel

### PR DESCRIPTION
**Pull request by easton-morris**
carousel

### CODE REVIEW

File: javascript-carousel/main.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+  for (let i = 0; i < (imageSet.length - 1); i++) {
+    const nextIcon = document.createElement('i');
+    nextIcon.setAttribute('class', 'progress-icon hoverable far fa-gem');
+    nextIcon.setAttribute('data-pos', (i + 1));
+    $progArea.appendChild(nextIcon);
+  }
+}
+
+document.addEventListener('load', createIcons());
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js:79]: Calling `createIcons` here means it won't wait for the `'load'` event to fire.

File: javascript-carousel/main.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      $iconList[0].setAttribute('class', 'progress-icon hoverable fas fa-gem');
+      $caroImg.setAttribute('src', imageSet[0].path);
+      $caroImg.setAttribute('alt', imageSet[0].altText);
+      $iconList[currentPos].setAttribute('class', 'progress-icon hoverable far fa-gem');
+      $caroImg.setAttribute('data-num', 0);
+    } else {
+      advanceImg();
+    }
+  }, 3 * 1000);
+}
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js:63]: Do you need really need two chunks of code that start an interval that does the same thing?

File: javascript-carousel/main.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  const progPos = event.target.getAttribute('data-pos');
+  $caroImg.setAttribute('src', imageSet[progPos].path);
+  $caroImg.setAttribute('alt', imageSet[progPos].altText);
+  const $resetIcon = document.querySelector('i[class="progress-icon hoverable fas fa-gem"]');
+  $resetIcon.setAttribute('class', 'progress-icon hoverable far fa-gem');
+  event.target.setAttribute('class', 'progress-icon hoverable fas fa-gem');
+  $caroImg.setAttribute('data-num', progPos);
+  window.clearInterval(intervalNum);
+  delayTimer();
+}
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js:128]: It seems like there's a lot of shared functionality between these three event handlers that could be broken out into its own function.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: javascript-carousel/main.js (hunk 2 above)
Comment (easton-morris) [reply javascript-carousel/main.js:63]: fixed: just called the function on load instead.

File: javascript-carousel/main.js (hunk 1 above)
Comment (easton-morris) [reply javascript-carousel/main.js:79]: fixed: removed the call, fed the callback instead.

Comment (easton-morris) [reply javascript-carousel/main.js:79]: just kidding, that was dumb. Just called it and ditched the event listener lol.

File: javascript-carousel/main.js (hunk 3 above)
Comment (easton-morris) [reply javascript-carousel/main.js:128]: broke out the updating portion into a function (updateItems) and called it where needed.

**Comment by easton-morris** [comment]
gif is the same again though...

**Review by TimelikeClosure — APPROVED** [review]
These commit messages are for your own reference to understand the changes made in each commit. Going forward, I highly suggest noting _what_ changed, rather than _why_ it changed.

---

image

**Pull request merged**